.github/workflows/kotlin-publish.yml

```diff
@@ -0,0 +1,121 @@
+name: Publish Kotlin to Maven Central
+
+on:
+  # TODO: remove after testing
+  push:
+    branches: [ci/kotlin-release]
+  release:
+    types: [published]
+
+jobs:
+  build:
+    # if: contains(github.event.release.tag_name, 'kotlin')
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: x86_64-unknown-linux-gnu
+            lib_prefix: lib
+            lib_ext: .so
+          - os: ubuntu-latest
+            target: aarch64-unknown-linux-gnu
+            lib_prefix: lib
+            lib_ext: .so
+          - os: macos-latest
+            target: x86_64-apple-darwin
+            lib_prefix: lib
+            lib_ext: .dylib
+          - os: macos-latest
+            target: aarch64-apple-darwin
+            lib_prefix: lib
+            lib_ext: .dylib
+          - os: windows-latest
+            target: x86_64-pc-windows-msvc
+            lib_prefix: ""
+            lib_ext: .dll
+          - os: windows-latest
+            target: aarch64-pc-windows-msvc
+            lib_prefix: ""
+            lib_ext: .dll
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@dded0888837ed1f317902acf8a20df0ad188d165 # v5.0.0
+        with:
+          java-version: "21"
+          distribution: "temurin"
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@0f44b27771c32bda9f458f75a1e241b09791b331 # v1.91.1
+        with:
+          targets: ${{ matrix.target }}
+          toolchain: stable
+
+      - name: Install cross-compilation tools
+        if: matrix.target == 'aarch64-unknown-linux-gnu'
+        run: sudo apt-get update && sudo apt-get install -y gcc-aarch64-linux-gnu
```

Comment:
I'm only installing `x86_64-unknown-linux-gnu` but now that I think about it, it's maybe not necessary 🤔